_LowLevelCode/cpp/cpp_communicator/cpp_communicator.cpp

```diff
@@ -213,14 +214,26 @@ void set_numlab_and_nlabs(class_handle<MPI_wrapper> * const pCommunicatorHolder,
             *pNlab = 0;
         }
     }
-    if (nlhs == (int)labIndex_Out::n_workers + 1) {
+    if (nlhs >= (int)labIndex_Out::n_workers + 1) { // also return NumLabs
         plhs[(int)labIndex_Out::n_workers] = mxCreateNumericMatrix(1, 1, mxINT32_CLASS, mxREAL);
         uint64_t* pNworkers = (uint64_t*)mxGetData(plhs[(int)labIndex_Out::n_workers]);
         if (pCommunicatorHolder) {
-            *pNworkers = (uint64_t)(uint64_t)pCommunicatorHolder->class_ptr->numLabs;
+            *pNworkers = (uint64_t)pCommunicatorHolder->class_ptr->numLabs;
         }
         else {
             *pNworkers = 0;
         }
     }
+    if (nlhs == (int)labIndex_Out::pool_names + 1) { // also return the names of the pool nodes
```

Comment:
`pool_names` is enumeration describing position of the pool_names in `lhs` array. If I place 2 here it would be worth. 
the position in C-arrays starts from 0, so number of arguments is n-total-positions +1. This is what checked. 
Converting enumeration into number in C++ causes warning. Its the specifics of C<->C++ interface.